src/main/java/moe/nightfall/vic/integratedcircuits/client/gui/component/GuiIOMode.java

```diff
@@ -29,7 +29,7 @@ public GuiIOMode(int id, int xPos, int yPos, GuiCAD parent, int side) {
 	public boolean mousePressed(Minecraft mc, int x, int y) {
 		boolean b = parent.blockMouseInput ? false : super.mousePressed(mc, x, y);
 		if (b)
-			parent.tileentity.setInputMode(side, EnumConnectionType.values()[(mode.ordinal() + 1) % 3]);
+			parent.tileentity.setInputMode(side, EnumConnectionType.values()[(mode.ordinal() + 1) % 4]);
```

Comment:
I am going to use another system for this anyway so that IO width can be taken into account...